### PR TITLE
ZBUG-3245 fallback pdf preview to browser pdf viewer and use iframe sandbox attribute

### PR DESCRIPTION
**Pull request by sitaram-mulik-synacor**
- Added implementation to open pdf document preview in seperate browser window.

### CODE REVIEW

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js [hunk 1]
```diff
@@ -824,6 +819,26 @@ function(){
...
+ZmPreviewView.prototype._handleVulnerableIframeContent =
+function(item, restUrl){
+	if(item.contentType === 'application/pdf') {
+		this._iframePreview.getIframe().setAttribute('sandbox', '');
+		this._renderPreviewLink(ZmMsg.previewNewTabLink, restUrl);
+	} else {
+		this._iframePreview.getIframe().removeAttribute('sandbox');
+		this._iframePreview.setSrc(restUrl);
+	}
+}
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js:831]:
```suggestion
};
```

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js [hunk 2]
```diff
@@ -824,6 +819,26 @@ function(){
...
+
+ZmPreviewView.prototype._renderPreviewLink = 
+function(msg, link){
+	this._iframePreview.setIframeContent([
+		"<div style='height:100%;width:100%;text-align:center;vertical-align:middle;padding-top:30px;font-family: \'Helvetica Neue\',Helvetica,Arial,\'Liberation Sans\',sans-serif;'>",
+		AjxMessageFormat.format(msg, link),
+		"</div>"
+	].join(''))
+}
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js:840]:
```suggestion
};
```

File: WebRoot/messages/ZmMsg_ar.properties [hunk 3]
```diff
@@ -2701,6 +2701,7 @@ presentation = \u0639\u0631\u0636 \u062a\u0642\u062f\u064a\u0645\u064a
 presentationName = \u0627\u0633\u0645 \u0627\u0644\u0639\u0631\u0636 \u0627\u0644\u062a\u0642\u062f\u064a\u0645\u064a
 preview = \u0645\u0639\u0627\u064a\u0646\u0629
 previewDownloadLink = \u064a\u062a\u0639\u0630\u0631 \u0639\u0644\u064a\u0646\u0627 \u0625\u0646\u0634\u0627\u0621 \u0645\u0639\u0627\u064a\u0646\u0629 \u0644\u0644\u0645\u0644\u0641. \u0627\u0646\u0642\u0631 \u0647\u0646\u0627 \u0645\u0646 \u0623\u062c\u0644 <a href="{0}" target="_blank" >\u062a\u0646\u0632\u064a\u0644</a> \u0627\u0644\u0645\u0639\u0627\u064a\u0646\u0629.\u200f
+previewNewTabLink = \u0644\u0623\u0633\u0628\u0627\u0628 \u0623\u0645\u0646\u064a\u0629 \u060c \u064a\u0631\u062c\u0649 <a href="{0}" target="_blank" >\u0627\u0644\u0646\u0642\u0631 \u0647\u0646\u0627</a> \u0644\u0639\u0631\u0636\u0647\u0627 \u0641\u064a \u0639\u0644\u0627\u0645\u0629 \u062a\u0628\u0648\u064a\u0628 \u062c\u062f\u064a\u062f\u0629.
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/messages/ZmMsg_ar.properties:2704]: Translation is missing for some languages (`en_AU`, `en_GB`, etc). Do we need to add it?

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js [hunk 4]
```diff
@@ -824,6 +819,26 @@ function(){
 };
 
 
+ZmPreviewView.prototype._handleVulnerableIframeContent =
+function(item, restUrl){
+	if(item.contentType === 'application/pdf') {
```
Comment (k-kato) [inline WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js:824]:
```suggestion
	if (item.contentType === 'application/pdf') {
```

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js [hunk 5]
```diff
@@ -824,6 +819,26 @@ function(){
 };
 
 
+ZmPreviewView.prototype._handleVulnerableIframeContent =
```
Comment (k-kato) [inline WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js:822]:
```suggestion
ZmPreviewView.prototype._handleIframeContent =
```

File: WebRoot/messages/ZmMsg_ar.properties (hunk 3 above)
Comment (sitaram-mulik-synacor) [reply WebRoot/messages/ZmMsg_ar.properties:2704]: en_AU and en_GB will fallback to the default labels present in ZmMsg.properties which are in English. Which I think is expected, had a discussion regarding same with @miteshsavani810 as well. Let me know if anything else is missing.

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js (hunk 1 above)
Comment (sitaram-mulik-synacor) [reply WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js]: Fixed.

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js (hunk 2 above)
Comment (sitaram-mulik-synacor) [reply WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js]: Fixed.

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js (hunk 4 above)
Comment (sitaram-mulik-synacor) [reply WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js]: Fixed.

File: WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js (hunk 5 above)
Comment (sitaram-mulik-synacor) [reply WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js]: Fixed.

**Review by k-kato — APPROVED** [review]

**Review by HimanshuSojitraSynacor — APPROVED** [review]

**Pull request merged**